### PR TITLE
OC-11782:Casebook PDF randomly fails to print Events in the order

### PR DESCRIPTION
**Pull request by Taoli2018**

### CODE REVIEW

File: src/main/java/core/org/akaza/openclinica/service/StudyParticipantServiceImpl.java [hunk 1]
```diff
@@ -461,6 +464,29 @@ public void startCaseBookPDFJob(JobDetail jobDetail,
...
+			    		for(EventDefinitionCrf edfc:edfcs) {
+			    			CrfBean crf = edfc.getCrf();
+			    			
+			    			if(eventCrf.getCrfVersion().getCrf().getCrfId() == crf.getCrfId()) {
+			    				eventCrf.setOrdinal(edfc.getOrdinal());
+			    			}
+			    		}			    		
+			    	}
+			    	
+			    	Comparator<EventCrf> compareByOrinal = new Comparator<EventCrf>() {
```
Comment (svadla-oc) [inline src/main/java/core/org/akaza/openclinica/service/StudyParticipantServiceImpl.java]: make this a static variable and move this comparator to EventCrf.java. Update the comparison logic such that you can fetch the ordinal directly from EvenCrf. 
Please check with Joe to find out if we should use FormLayout or CrfVersion

Comment (Taoli2018) [reply src/main/java/core/org/akaza/openclinica/service/StudyParticipantServiceImpl.java]: done

File: src/main/java/core/org/akaza/openclinica/domain/datamap/EventCrf.java [hunk 2]
```diff
@@ -58,7 +60,8 @@
     private Integer sdvUpdateId;
     private List<DnEventCrfMap> dnEventCrfMaps;
     private List<ItemData> itemDatas;
-
+    
+    static Comparator<EventCrf> compareByOrinal;
```
Comment (svadla-oc) [inline src/main/java/core/org/akaza/openclinica/domain/datamap/EventCrf.java]: ordinal is misspelled in this file.

Comment (Taoli2018) [reply src/main/java/core/org/akaza/openclinica/domain/datamap/EventCrf.java]: done

File: src/main/java/core/org/akaza/openclinica/domain/datamap/EventCrf.java [hunk 3]
```diff
@@ -356,5 +359,47 @@ public FormLayout getFormLayout() {
...
+			        return o1ordinal.compareTo(o1ordinal);
+			    };
+                 
+			    /**
+				 * @param ec
+				 * @return 
+				 */
+				private Integer getOrdinal(EventCrf ec) {
+					Integer ecOrdinal = null;
+					List<EventDefinitionCrf> edfcList1 = ec.getFormLayout().getCrf().getEventDefinitionCrfs();
```
Comment (svadla-oc) [inline src/main/java/core/org/akaza/openclinica/domain/datamap/EventCrf.java]: Let's test this logic with two events, Event1 & Event2 with forms 1, 2 & 3 with different ordering in Event1 vs Event2.

File: src/main/java/core/org/akaza/openclinica/domain/datamap/EventCrf.java [hunk 4]
```diff
@@ -356,5 +359,47 @@ public FormLayout getFormLayout() {
     public void setFormLayout(FormLayout formLayout) {
         this.formLayout = formLayout;
     }
+   
+
+	public static Comparator<EventCrf> getCompareByOrdinal() {
```
Comment (svadla-oc) [inline src/main/java/core/org/akaza/openclinica/domain/datamap/EventCrf.java]: fix identation

File: src/main/java/core/org/akaza/openclinica/domain/datamap/EventCrf.java (hunk 3 above)
Comment (Taoli2018) [reply src/main/java/core/org/akaza/openclinica/domain/datamap/EventCrf.java]: done

File: src/main/java/core/org/akaza/openclinica/domain/datamap/EventCrf.java (hunk 4 above)
Comment (Taoli2018) [reply src/main/java/core/org/akaza/openclinica/domain/datamap/EventCrf.java]: done

**Comment by svadla-oc** [comment]
👍

**Pull request merged**